dbt/include/databricks/macros/relations/components/tblproperties.sql

```diff
@@ -1,4 +1,5 @@
 {% macro get_create_sql_tblproperties(tblproperties) %}
+  {%- set _ = adapter.check_iceberg(config.get('table_format'), config.get('file_format'), model.config.materialized) -%}
```

Comment:
Here we check and fail the compilation of incremental tblproperties if we are misconfigured.  Otherwise, since this is used only for incremental, the tblproperties will already have the iceberg properties included.